crates/router/src/core/payouts/helpers.rs

```diff
@@ -194,6 +194,43 @@ pub async fn make_payout_method_data<'a>(
     }
 }
 
+pub fn should_create_connector_transfer_method(
+    payout_data: &PayoutData,
+    connector_data: &api::ConnectorData,
+) -> RouterResult<Option<String>> {
+    let mut connector_mandate_id = None;
+    #[cfg(all(
+        any(feature = "v2", feature = "v1"),
+        not(feature = "payment_methods_v2")
+    ))]
+    if let Some(pm) = &payout_data.payment_method {
+        let connector_mandate_details = pm
+            .connector_mandate_details
+            .clone()
+            .map(|details| {
+                details.parse_value::<diesel_models::PaymentsMandateReference>(
+                    "connector_mandate_details",
+                )
+            })
+            .transpose()
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("unable to deserialize connector mandate details")?;
+
+        if let Some(merchant_connector_id) = connector_data.merchant_connector_id.as_ref() {
+            if let Some(mandate_reference_record) = connector_mandate_details
+                .clone()
+                .get_required_value("connector_mandate_details")
```

Comment:
This should not be a required field

In case `connector_mandate_details` don't exist, we should return `None` in this function.

Reply:
done